cql3/expr/expression.cc

```diff
@@ -1265,6 +1222,67 @@ expression replace_token(const expression& expr, const column_definition* new_cd
         }, expr);
 }
 
+expression search_and_replace(const expression& e,
+        const noncopyable_function<std::optional<expression> (const expression& candidate)>& replace_candidate) {
+    //auto recurse = std::bind(search_and_replace, std::placeholders::_1, std::ref(replace_candidate));
+    auto recurse = [&] (const expression& e) -> expression {
+        return search_and_replace(e, replace_candidate);
+    };
+    return replace_candidate(e).value_or(
+            std::visit(overloaded_functor{
+                [] (bool b) -> expression { return b; },
```

Comment:
`bool` qualifies as a `LeafExpression`, it could be handled with other `LeafExpressions`

Reply:
Yes